controllers/configurationpolicy_controller.go

```diff
@@ -932,6 +932,23 @@ func sortRelatedObjectsAndUpdate(
 		}
 	}
 
+	// add RelatedObject into newRelated which is in oldRelated but not in newRelated
+	for _, oldEntry := range oldRelated {
+		ifAppend := true
+
+		if oldEntry.Properties != nil && (*oldEntry.Properties.CreatedByPolicy) {
```

Comment:
If the delete option is ever `DeleteAll`, then it wouldn't matter if the object was created by the policy or not, right? So I'm wondering if this `if` statement should be included at all.

Reply:
If we still decide to fix the original issue, yes, need to remove this and keep all objects in the `relatedObjects`, then determined by `DeleteAll` or `DeleteIfCreated` in `cleanUpChildObjects`